ufo2fontir/src/source.rs

```diff
@@ -193,21 +203,20 @@ impl Source for DesignSpaceIrSource {
                 if !glif_file.exists() {
                     return Err(Error::FileExpected(glif_file));
                 }
+                glyph_names.insert(glyph_name.clone());
```

Comment:
here you are collecting the union of all the glyph names across all the source UFOs, irrespective of them being present or not in the default source, but as I think we discussed only glyphs that are in the default source will exist in the output VF, the non-default sources may contain _less_ glyphs (be sparse), but not _more_ glyphs than the default one.
Since all VF font projects are designed like that it's unlikely that we'll ever find in such a situation where the nomially default source does not already contain the superset of all glyphs defined across all sources.
But I figured I should mention this here. And perhaps we could even add an explicit check for it, skip + warn if there's any such glyph defined elsewhere than default source.

Reply:
Oh good catch. I really hate the idea of throwing away data but I suppose it's correct here.

Reply:
Added a test to demonstrate the problem and fixed